redbeams/src/main/java/com/sequenceiq/redbeams/sync/DBStackStatusSyncService.java

```diff
@@ -0,0 +1,94 @@
+import org.springframework.stereotype.Component;
+
+import javax.inject.Inject;
+
+import com.sequenceiq.cloudbreak.cloud.CloudConnector;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+import com.sequenceiq.cloudbreak.cloud.context.CloudContext;
+import com.sequenceiq.cloudbreak.cloud.init.CloudPlatformConnectors;
+import com.sequenceiq.cloudbreak.cloud.model.CloudCredential;
+import com.sequenceiq.cloudbreak.cloud.model.ExternalDatabaseStatus;
+import com.sequenceiq.cloudbreak.cloud.model.Location;
+import com.sequenceiq.redbeams.api.model.common.DetailedDBStackStatus;
+import com.sequenceiq.redbeams.api.model.common.Status;
+import com.sequenceiq.redbeams.converter.cloud.CredentialToCloudCredentialConverter;
+import com.sequenceiq.redbeams.domain.stack.DBStack;
+import com.sequenceiq.redbeams.dto.Credential;
+import com.sequenceiq.redbeams.service.CredentialService;
+import com.sequenceiq.redbeams.service.stack.DBStackStatusUpdater;
+
+import static com.sequenceiq.cloudbreak.cloud.model.AvailabilityZone.availabilityZone;
+import static com.sequenceiq.cloudbreak.cloud.model.Location.location;
+import static com.sequenceiq.cloudbreak.cloud.model.Region.region;
+
+@Component
+public class DBStackStatusSyncService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DBStackStatusSyncService.class);
+
+    @Inject
+    private CredentialService credentialService;
+
+    @Inject
+    private CredentialToCloudCredentialConverter credentialConverter;
+
+    @Inject
+    private CloudPlatformConnectors cloudPlatformConnectors;
+
+    @Inject
+    private DBStackStatusUpdater dbStackStatusUpdater;
+
+    public void sync(DBStack dbStack) {
+        DetailedDBStackStatus detailedDBStackStatus = getDetailedDBStackStatusFromProvider(dbStack);
+        Status status = detailedDBStackStatus.getStatus();
+        if (!dbStack.getStatus().equals(status)) {
+            LOGGER.debug(":::Auto sync::: Update DB Stack Status from '{}' to '{}'", dbStack.getStatus(), status);
+
+            dbStackStatusUpdater.updateStatus(dbStack.getId(), detailedDBStackStatus);
+        }
+    }
+
+    private DetailedDBStackStatus getDetailedDBStackStatusFromProvider(DBStack dbStack) {
+        ExternalDatabaseStatus externalDatabaseStatus = getExternalDatabaseStatus(dbStack);
+        DetailedDBStackStatus detailedDBStackStatus = null;
+        switch (externalDatabaseStatus) {
+            case STARTED: detailedDBStackStatus = DetailedDBStackStatus.STARTED;
+                break;
+            case STOPPED: detailedDBStackStatus = DetailedDBStackStatus.STOPPED;
+                break;
+            case STOP_IN_PROGRESS: detailedDBStackStatus = DetailedDBStackStatus.STOP_IN_PROGRESS;
+                break;
+            case START_IN_PROGRESS: detailedDBStackStatus = DetailedDBStackStatus.START_IN_PROGRESS;
+                break;
+            default: detailedDBStackStatus = DetailedDBStackStatus.UNKNOWN;
```

Comment:
is `STARTED ` means `AVAILABLE`? What is the difference between `STARTED` and `AVAILABLE`?

Reply:
Yes, STARTED means AVAILABLE but in a different context. As far as I can see AVAILABLE is used by legacy reasons.

Reply:
if in STARTED state, the ui show the db as available?

Reply:
I have not created new status messages for the UI so if it is shown up "available" after the creation, it should be "available" when the start operation is finished.

Reply:
Where do we display the status on the UI?
Anyway, the status in the DB is AVAILABLE both after start and create, but after start the detaileddbstackstatus is STARTED.

Reply:
Oh, I see.